lib/util/git/index.ts

```diff
@@ -418,9 +418,21 @@ export async function syncGit(): Promise<void> {
 }
 
 // istanbul ignore next
-export async function getRepoStatus(): Promise<StatusResult> {
+export async function getRepoStatus(path?: string): Promise<StatusResult> {
+  if (path !== undefined) {
+    const { localDir } = GlobalConfig.get();
+    const localPath = upath.resolve(localDir, path);
+    if (!localPath.startsWith(upath.resolve(localDir))) {
+      logger.warn(
+        { localPath, localDir },
+        'Preventing access to file outside the local directory'
+      );
+      return Promise.reject();
```

Comment:
```suggestion
      throw new Error(INVALID_PATH);
```

`INVALID_PATH = 'invalid-path'` should be added to `lib/constants/error-messages.ts`

@rarkins Is this naming ok?